src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -49,6 +49,17 @@ class LedgerRPC_test : public beast::unit_test::suite
             BEAST_EXPECT(jv[jss::error_message] == msg);
     };
 
+    // Corrupt a valid address by replacing the 10th character with '!'.
+    // '!' is not part of the ripple alphabet.
+    std::string
+    makeBadAddress (std::string const& good)
+    {
+        std::string ret = good;
+        ret.replace (10, 1, 1, '!');
```

Comment:
Rather than change it to an invalid character, how about changing it to a valid character and relying on the check sum to flag this as a bad address (the are four bytes in the checksum, so it should do so with a high enough probability for a unit test).

Reply:
This is only one way in which an address can be bad. We should (deterministically) fuzz addresses in multiple ways.

For example: permute pairs of characters, trim the string by removing each character, replacing each character (not including the first) and using an incorrect prefix, include invalid characters, etc.

Reply:
Just to put some perspective on this, `makeBadAddress()` is not intended as a general test for bad addresses.

In order to exercise some of the error handling code in `LedgerEntry.cpp` I need to supply a bad address as a parameter.  True, this is only one of many ways to produce a bad address.  But, in order to exercise `LedgerEntry.cpp`, I don't need a general mechanism for providing a bad address.  I just need something that looks enough like a valid address to slip all the way into `LedgerEntry.cpp` where the error can be detected and reacted to.  The technique employed worked well enough for the purposes of this test.

IMO, a full test that fuzzes bad addresses would belong in a different file.

Reply:
Agreed, @scottschurr. Thanks for the clarification.